### PR TITLE
Update licensing

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

This PR isolates the eBPF programs into a separate folder and loads them from the filesystem instead of bundling them into the agent binary.

### Motivation

Clearly separate the GPL parts from the rest of the agent.

### CODE REVIEW

File: tasks/go.py [hunk 1]
```diff
@@ -479,7 +479,7 @@ def reset(ctx):
 
 
 @task
-def generate(ctx):
+def generate(ctx, build_tags=""):
```
Comment (mx-psi) [inline tasks/go.py:482]: #5725 changed the way build tags are managed, here and in other places you are reverting to the old behavior. You should merge master into this branch so that this doesn't happen

File: pkg/ebpf/bytecode/asset_nobindata.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"path"
+
+	"github.com/pkg/errors"
+)
+
+// GetReader returns a new AssetReader for the specified file asset
+func GetReader(name string) (AssetReader, error) {
+	asset, err := os.Open(path.Join(DefaultBPFDir, path.Base(name)))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not find asset")
```
Comment (leeavital) [inline pkg/ebpf/bytecode/asset_nobindata.go:16]: Does the error include the path that we searched at? That might be useful for debugging bad configurations

Comment (lebauce) [reply pkg/ebpf/bytecode/asset_nobindata.go:16]: Yes, it does

File: pkg/process/config/yaml_config.go [hunk 3]
```diff
@@ -61,6 +62,10 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 	}
 
 	a.SysProbeBPFDebug = config.Datadog.GetBool(key(spNS, "bpf_debug"))
+	if config.Datadog.IsSet(key(spNS, "bpf_dir")) {
+		bytecode.DefaultBPFDir = config.Datadog.GetString(key(spNS, "bpf_dir"))
+	}
```
Comment (leeavital) [inline pkg/process/config/yaml_config.go:67]: This is inconsistent with how config is managed through the rest of the system probe. Usually, we add a field on `config` and pass it in where it's needed.

Comment (lebauce) [reply pkg/process/config/yaml_config.go:67]: True. I did try this approach first. But it implies passing this configuration setting in many places : tracer, oom, tcp queue length and security agent. I can go this way though

File: tasks/system_probe.py [hunk 4]
```diff
@@ -173,18 +176,20 @@ def test(ctx, skip_object_files=False, only_check_bpf_bytes=False):
...
-        build_object_files(ctx, install=False)
+        build_object_files(ctx, bundle_ebpf=bundle_ebpf)
 
     pkg = "./pkg/ebpf/... ./pkg/network/..."
 
     # Pass along the PATH env variable to retrieve the go binary path
     path = os.environ['PATH']
 
-    cmd = 'go test -mod={go_mod} -v -tags "{bpf_tag}" {pkg}'
+    cmd = 'go test -mod= -v -tags "{bpf_tag}" {pkg}'
```
Comment (leeavital) [inline tasks/system_probe.py:186]: why this change ?

File: omnibus/config/software/system-probe.rb [hunk 5]
```diff
@@ -13,4 +13,15 @@
...
+  mkdir "#{install_dir}/embedded/share/system-probe/ebpf"
+
+  copy 'pkg/ebpf/c/COPYING', "#{install_dir}/embedded/share/system-probe/ebpf/"
+  copy 'pkg/ebpf/bytecode/tracer-ebpf.o', "#{install_dir}/embedded/share/system-probe/ebpf/"
+  copy 'pkg/ebpf/bytecode/tracer-ebpf-debug.o', "#{install_dir}/embedded/share/system-probe/ebpf/"
+  copy 'pkg/ebpf/c/oom-kill-kern.c', "#{install_dir}/embedded/share/system-probe/ebpf/"
+  copy 'pkg/ebpf/c/tcp-queue-length-kern.c', "#{install_dir}/embedded/share/system-probe/ebpf/"
+
+  copy 'pkg/security/ebpf/c/runtime-security.o', "#{install_dir}/embedded/share/system-probe/ebpf/"
+  copy 'pkg/security/ebpf/c/runtime-security-syscall-wrapper.o', "#{install_dir}/embedded/share/system-probe/ebpf/"
```
Comment (leeavital) [inline omnibus/config/software/system-probe.rb:26]: Do we need to chmod these so they're owned by root ?

File: pkg/ebpf/bytecode/asset_bindata.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+// GetReader returns a new AssetReader for the specified bundled asset
+func GetReader(dir, name string) (AssetReader, error) {
+	content, err := Asset(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes.NewReader(content), nil
+}
```
Comment (leeavital) [inline pkg/ebpf/bytecode/asset_bindata.go:17]: Is it worth maintaining two builds with and without bindata? 

We're not going to distribute both, and it's not like users can switch between the two with a flag.

Comment (brycekahle) [reply pkg/ebpf/bytecode/asset_bindata.go:17]: IMHO yes. Otherwise testing internally would require an omnibus build, correct?

File: omnibus/config/software/system-probe.rb (hunk 5 above)
Comment (lebauce) [reply omnibus/config/software/system-probe.rb:26]: I don't know much about omnibus but it seems that `chown -R dd-agent:dd-agent /opt/datadog-agent` is done in `omnibus/package-scripts/agent/postinst`

File: tasks/system_probe.py (hunk 4 above)
Comment (lebauce) [reply tasks/system_probe.py:186]: Good catch. It should not have been commited

File: pkg/ebpf/bytecode/asset_bindata.go (hunk 6 above)
Comment (lebauce) [reply pkg/ebpf/bytecode/asset_bindata.go:17]: For the runtime security agent, we are compiling the functional tests into a binary `go test -c -o`. It would help a lot to have everything in one binary

Comment (lebauce) [reply pkg/ebpf/bytecode/asset_bindata.go:17]: Keeping the bindata also allows us to keep using the `TestEbpfBytesCorrect` to ensure the compiled versions are in sync with the sources

File: omnibus/config/software/system-probe.rb (hunk 5 above)
Comment (albertvaka) [reply omnibus/config/software/system-probe.rb:26]: Would it be better for them to be owned by root? We can do it by changing the `postinst` script as @lebauce suggested, but I'm not sure we need that.

**Review by leeavital — APPROVED** [review]

File: pkg/ebpf/bytecode/asset_bindata.go (hunk 6 above)
Comment (brycekahle) [reply pkg/ebpf/bytecode/asset_bindata.go:17]: @leeavital we deploy docker containers to clusters for testing quite often that don't have the source tree. If we don't have bundling, those Dockerfiles would need to be updated appropriately.

**Review by derekwbrown — APPROVED** [review]
Seems fine to me.
